File: dev-docs/snippets/documents/in-sync.swift
import JsBaoClient

// Check whether client and server hold identical document state via an async
// WebSocket state-vector round-trip. Resolves false when offline or on timeout.
func inSync(client: JsBaoClient, documentId: String) async {
  // #region example
  let synced = await client.documents.inSync(documentId: documentId)
  // #endregion example
  _ = synced
}

File: dev-docs/snippets/documents/in-sync.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Check whether client and server hold identical document state.
export async function inSync(client: JsBaoClient, documentId: string) {
  // #region example
  const synced = await client.documents.inSync(documentId);
  // #endregion example
  return synced;
}
